dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteRunnable.java

```diff
@@ -741,7 +742,11 @@ public WorkflowSubmitStatue call() {
             return WorkflowSubmitStatue.SUCCESS;
         } catch (Exception e) {
             log.error("Start workflow error", e);
-            return WorkflowSubmitStatue.FAILED;
+            if (e instanceof SQLException) {
+                return WorkflowSubmitStatue.FAILED;
+            } else {
+                return WorkflowSubmitStatue.ERROR;
+            }
```

Comment:
I suggest not to distinguish these two types of error, rather, DB failure will keep writing logs.